velox/exec/Window.cpp

```diff
@@ -282,6 +342,94 @@ void Window::callResetPartition(vector_size_t partitionNumber) {
   }
 }
 
+void Window::updateFrameBounds(
+    const vector_size_t& i,
+    const bool& isStartBound,
+    const vector_size_t& startRow,
+    const vector_size_t& numRows) {
+  auto firstPartitionRow = partitionStartRows_[currentPartition_];
+  auto lastPartitionRow = partitionStartRows_[currentPartition_ + 1] - 1;
+  auto rawPeerStarts = peerStartBuffer_->asMutable<vector_size_t>();
+  auto rawPeerEnds = peerEndBuffer_->asMutable<vector_size_t>();
+  auto rawFrameBounds = isStartBound
+      ? frameStartBuffers_[i]->asMutable<vector_size_t>()
+      : frameEndBuffers_[i]->asMutable<vector_size_t>();
+  auto type = windowFrames_[i].type;
+  auto boundType =
+      isStartBound ? windowFrames_[i].startType : windowFrames_[i].endType;
+  auto frameArg = isStartBound ? windowFrames_[i].start : windowFrames_[i].end;
+
+  auto updateKRowsFrameBounds = [&](const bool isKPreceding,
```

Comment:
On second thoughts updateKRowsFrameBounds can be a separate function.

Reply:
Updated, thanks!